src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -463,39 +516,43 @@ void func_80AE3260(EnRd* this, GlobalContext* globalCtx) {
     }
 }
 
-void func_80AE33F0(EnRd* this) {
-    Animation_PlayOnce(&this->skelAnime, &object_rd_Anim_004ADC);
-    this->unk_30C = this->unk_304 = 0;
-    this->unk_319 = 200;
-    this->unk_31B = 8;
+void EnRd_SetupGrab(EnRd* this) {
+    Animation_PlayOnce(&this->skelAnime, &gGibdoRedeadGrabStartAnim);
+    this->timer.animationJudder = this->grabState = 0;
+    this->grabDamageTimer = 200;
+    this->action = EN_RD_ACTION_GRAB;
     this->actor.speedXZ = 0.0f;
-    EnRd_SetupAction(this, func_80AE3454);
+    EnRd_SetupAction(this, EnRd_Grab);
 }
 
-void func_80AE3454(EnRd* this, GlobalContext* globalCtx) {
+void EnRd_Grab(EnRd* this, GlobalContext* globalCtx) {
     s32 pad;
     Player* player = GET_PLAYER(globalCtx);
 
     if (SkelAnime_Update(&this->skelAnime)) {
-        this->unk_304++;
+        this->grabState++;
     }
 
-    switch (this->unk_304) {
-        case 1:
-            Animation_PlayLoop(&this->skelAnime, &object_rd_Anim_004268);
-            this->unk_304++;
+    switch (this->grabState) {
+        case EN_RD_GRAB_INITIAL_DAMAGE:
+            Animation_PlayLoop(&this->skelAnime, &gGibdoRedeadGrabAttackAnim);
+            this->grabState++;
             globalCtx->damagePlayer(globalCtx, -8);
             func_800AA000(this->actor.xzDistToPlayer, 0xFF, 1, 0xC);
-            this->unk_319 = 20;
-        case 0:
-            Math_SmoothStepToS(&this->unk_30E, 0, 1, 0x5DC, 0);
-            Math_SmoothStepToS(&this->unk_310, 0, 1, 0x5DC, 0);
-        case 2:
+            this->grabDamageTimer = 20;
+            // fallthrough
+
+        case EN_RD_GRAB_START:
+            Math_SmoothStepToS(&this->headYRotation, 0, 1, 0x5DC, 0);
+            Math_SmoothStepToS(&this->upperBodyYRotation, 0, 1, 0x5DC, 0);
+            // fallthrough
+
+        case EN_RD_GRAB_ATTACK:
```

Comment:
we havent standardized switch styling at all, but imo fallthroughs look nice without a line break between the cases. it gets the point across of the fall through much more. Then cases that break properly can have line breaks to help emphasize that fact